### PR TITLE
Add basic build/tests, and add lifeline

### PR DESCRIPTION
**Pull request by bloodearnest**
The tests are exactly the same as the python-docker tests

We also synchronise the packages between them, including the new
lifelines package

### CODE REVIEW

File: .github/workflows/publish_docker.yaml [hunk 1]
```diff
@@ -7,31 +7,15 @@ on:
...
-    runs-on: ubuntu-latest
-    name: Build and publish docker image
-    needs: tag-new-version
-    if: needs.tag-new-version.outputs.tag
+  build:
+    runs-on: ubuntu-20.04
     steps:
     - name: Checkout
-      uses: actions/checkout@v1
+      uses: actions/checkout@master
```
Comment (evansd) [inline .github/workflows/publish_docker.yaml]: This seems weird. I think it's OK not to pin to a sha for official actions like this, but don't we want to specify a version tag at least?

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+on:
+  pull_request:
+env:
+  IMAGE_NAME: jupyter
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
```
Comment (evansd) [inline .github/workflows/tests.yaml]: As above

File: tests/test_import.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            if not line or line.startswith("#"):
+                continue
+
+            name, _, _ = line.partition("==")
+            for module in get_module_names(name):
+                yield name, module
+
+
+@pytest.mark.parametrize(
+    "name, module", generate_import_names(Path("requirements.txt"))
```
Comment (evansd) [inline tests/test_import.py:41]: :+1: Glorious

**Review by evansd — APPROVED** [review]
Looks good, although I'm not totally on top of all the context here. One, non-blocking question about `actions/checkout@master`.

File: .github/workflows/publish_docker.yaml (hunk 1 above)
Comment (bloodearnest) [reply .github/workflows/publish_docker.yaml]: huh, I didn't do that one on purpose, will pin to @v2

**Pull request merged**